sdk/dotnet/Pulumi.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+using Xunit;
+
+namespace Pulumi.Tests.Serialization
+{
+    public class EnumConverterTests : ConverterTests
+    {
+        [EnumType]
+        public readonly struct ContainerColor : IEquatable<ContainerColor>
+        {
+            private readonly string _value;
+
+            private ContainerColor(string value)
+            {
+                _value = value ?? throw new ArgumentNullException(nameof(value));
+            }
+
+            public static ContainerColor Red { get; } = new ContainerColor("red");
+            public static ContainerColor Blue { get; } = new ContainerColor("blue");
+            public static ContainerColor Yellow { get; } = new ContainerColor("yellow");
+
+            public static bool operator ==(ContainerColor left, ContainerColor right) => left.Equals(right);
+            public static bool operator !=(ContainerColor left, ContainerColor right) => !left.Equals(right);
+
+            public static explicit operator string(ContainerColor value) => value._value;
```

Comment:
To get the value out of the struct, I decided to opt for an explicit conversion operator that converts the enum type to its underlying type (in this case `string`), rather than have a `Value` property (or something else).

This makes getting the underlying value consistent with normal enums that are built-in to the language, e.g.:

```c#
int four = (int)ContainerSize.FourInch;
string red = (string)ContainerColor.Red;
```